mathics/builtin/string/operations.py

```diff
@@ -604,7 +619,7 @@ def apply_n(self, string, patt, n, evaluation, options):
             overlap = False  # unknown options are teated as False
 
         # convert patterns
-        if patt.has_form("List", None):
+        if patt.get_head() is SymbolList:
```

Comment:
I have been leaving all of this because it is going to eventually get changed to `isinstance(ListExpression)`. 

Having the older form may make things easier to do a search and replace on.